pkg/rc/replication_controller.go

```diff
@@ -952,13 +953,28 @@ func (rc *replicationController) scheduleNewNodeForTransfer(rcFields fields.RC,
 func (rc *replicationController) doBackgroundNodeTransfer(rcFields fields.RC, logger logging.Logger) {
 	ok, rollbackReason := rc.watchHealth(rcFields, logger)
 	if ok {
-		err := rc.finishTransfer(rcFields, logger)
-		if err == nil {
-			logger.Infoln("Node transfer complete.")
-			return
+		// We retry finishTransfer() up to 10 times in case of a
+		// fleeting transaction building error or network error
+		var err error
+		for i := 0; i < 10; i++ {
+			err = rc.finishTransfer(rcFields, logger)
+			if err == nil {
+				logger.Infoln("Node transfer complete.")
+				return
+			}
+
+			rc.logger.WithError(err).Errorln("could not do final node transfer transaction, retrying")
+
+			backoff := time.Duration(math.Pow(2, float64(i))) * time.Second
+			time.Sleep(backoff)
+		}
+		rc.logger.WithError(err).Errorln("could not do final node transfer transaction after retries, rolling back")
+		alertErr := rc.alerter.Alert(rc.alertInfo(rcFields, err.Error()), alerting.LowUrgency)
+		if alertErr != nil {
+			logger.WithError(alertErr).Errorln("Unable to send alert")
 		}
 
-		rc.logger.WithError(err).Errorln("could not do final node transfer transaction, rolling back")
+		rc.nodeTransfer = nodeTransfer{}
```

Comment:
I think you need to protect this by grabbing `nodeTransferMu` or else there's a data race